src/pip/_internal/cli/req_command.py

```diff
@@ -195,7 +197,33 @@ def __init__(self, *args, **kw):
         self.cmd_opts.add_option(cmdoptions.no_clean())
 
     @staticmethod
+    def determine_resolver_variant(options):
+        # type: (Values) -> str
+        """Determines which resolver should be used, based on the given options."""
+        # We didn't want to change things for Python 2, since it's nearly done with
+        # and we're using performance improvements that only work on Python 3.
+        if PY2:
+            if '2020-resolver' in options.features_enabled:
+                return "2020-resolver"
+            else:
+                return "legacy"
+
+        # Warn about the options that are gonna be removed.
+        if '2020-resolver' in options.features_enabled:
```

Comment:
```suggestion
        if '2020-resolver' in options.features_enabled and not PY2:
```

?

Reply:
We never reach here with PY2 being True (all the branches on lines 205-209 end with an early return).